include/networkit/io/NetworkitBinaryGraph.hpp

```diff
@@ -28,5 +42,77 @@ static constexpr uint64_t DIR_MASK = 0x1; // bit 0
 static constexpr uint64_t WGHT_MASK = 0xE; //bit 1-3
 static constexpr uint64_t WGHT_SHIFT = 0x1;
 
+/// Serializes value into a buffer and returns the number of bytes written.
+inline size_t varIntEncode(uint64_t value, uint8_t *buffer) noexcept {
+	if (!value) {
+		buffer[0] = 1;
+		return 1;
+	}
+
+	const auto bits = 64 - tlx::clz(value);
+	auto dataBytes = bits / 8;
+
+	if (dataBytes < 8) {
+		// check whether the remaining value bits fit into remaining bits of buffer[0]
+		auto remain = bits % 8;
+		dataBytes += (remain + dataBytes >= 8);
+	}
```

Comment:
The computation of `dataBytes` can be simplified by dividing `bits` by 7, see my top-level comment.